landlab/components/sink_fill/sink_fill_barnes.py

```diff
@@ -0,0 +1,258 @@
+#!/usr/env/python
+
+"""
+fill_sinks_barnes.py
+
+Fill sinks in a landscape to the brim, following the Barnes et al. (2014) algos.
+"""
+
+from __future__ import print_function
+
+import warnings
+
+from landlab import FieldError, Component, BAD_INDEX_VALUE
+from landlab import RasterModelGrid, VoronoiDelaunayGrid  # for type tests
+from landlab.components import LakeMapperBarnes
+from landlab.utils.return_array import return_array_at_node
+from landlab.core.messages import warning_message
+from landlab.components.lake_fill.lake_fill_barnes import StablePriorityQueue
+
+from landlab import FIXED_VALUE_BOUNDARY, FIXED_GRADIENT_BOUNDARY
+from landlab import CLOSED_BOUNDARY
+from collections import deque
+import six
+import numpy as np
+import heapq
+import itertools
+
+LOCAL_BAD_INDEX_VALUE = BAD_INDEX_VALUE
+
+
+class SinkFillerBarnes(LakeMapperBarnes):
+    """
+    Uses the Barnes et al (2014) algorithms to replace pits in a topography
+    with flats, or optionally with very shallow gradient surfaces to allow
+    continued draining.
+
+    This component is NOT intended for use iteratively as a model runs;
+    rather, it is to fill in an initial topography. If you want to repeatedly
+    fill pits as a landscape develops, you are after the SinkFillerBarnes
+    component.
+
+    The locations and depths etc. of the fills will be tracked, and properties
+    are provided to access this information.
+
+    Parameters
+    ----------
+    grid : ModelGrid
+        A grid.
+    surface : field name at node or array of length node
+        The surface to fill.
+    method : {'steepest', 'd8'}
+        Whether or not to recognise diagonals as valid flow paths, if a raster.
+        Otherwise, no effect.
+    fill_flat : bool
+        If True, pits will be filled to perfectly horizontal. If False, the new
+        surface will be slightly inclined to give steepest descent flow paths
+        to the outlet.
```

Comment:
In the other component there is a lot of discussion around how this term impacts the topographic steepest slope field. Does that apply here? Is a user expected to re-run Flow Direction and/or Accumulation on this post-sink fill?

Reply:
Made this a bit clearer